Spigot-Server-Patches/0549-Fix-cartography-table.patch

```diff
@@ -0,0 +1,201 @@
+  Map locking now happens in-place (ItemWorldMap#lockMap(ItemStack)), meaning that the method does not return a new ItemStack.
+    This allows onTake(EntityHuman, ItemStack) to modify the output item in-place, which helps solve the second issue.
+    Locking in-place is also more consistent since scaling also happens in-place.
+  When the output slot is shift clicked and the map is being locked or scaled, the new map will only be merged in the player's inventory if they have an empty slot.
+    Merging in this case happens after onTake(EntityHuman, ItemStack) is called to allow the in-place locking to happen first.
+    Updating the player's inventory is sometimes necessary because the client thinks that something else happened.
+    This solves the second and third issues.
+  Obfhelpers have been added to makes these changes easier to understand.
+
+diff --git a/src/main/java/net/minecraft/server/Container.java b/src/main/java/net/minecraft/server/Container.java
+index 268bedd042b651514aa94b735876de9c40bae56f..b6d749941631e3f55ad7e06579419cc112d969e9 100644
+--- a/src/main/java/net/minecraft/server/Container.java
++++ b/src/main/java/net/minecraft/server/Container.java
+@@ -565,6 +565,7 @@ public abstract class Container {
+ 
+     public abstract boolean canUse(EntityHuman entityhuman);
+ 
++    protected boolean mergeItemStack(ItemStack itemstack, int i, int j, boolean flag) { return a(itemstack, i, j, flag); } // Paper - OBFHELPER
+     protected boolean a(ItemStack itemstack, int i, int j, boolean flag) {
+         boolean flag1 = false;
+         int k = i;
+diff --git a/src/main/java/net/minecraft/server/ContainerAccess.java b/src/main/java/net/minecraft/server/ContainerAccess.java
+index 6ba10d61a1825ff6a5b083d19176ce922fa36d56..3020a15b6e75db40b13ebb7919fcac5f47247d4b 100644
+--- a/src/main/java/net/minecraft/server/ContainerAccess.java
++++ b/src/main/java/net/minecraft/server/ContainerAccess.java
+@@ -54,6 +54,7 @@ public interface ContainerAccess {
+         return this.a(bifunction).orElse(t0);
+     }
+ 
++    default void consume(BiConsumer<World, BlockPosition> biconsumer) { this.a(biconsumer); } // Paper - OBFHELPER
+     default void a(BiConsumer<World, BlockPosition> biconsumer) {
+         this.a((world, blockposition) -> {
+             biconsumer.accept(world, blockposition);
+diff --git a/src/main/java/net/minecraft/server/ContainerCartography.java b/src/main/java/net/minecraft/server/ContainerCartography.java
+index 79d328786f2e9ba386cb297bb8e7ec0ec3228a65..edc90b8508e4c1d32070393be77f2654adac49e1 100644
+--- a/src/main/java/net/minecraft/server/ContainerCartography.java
++++ b/src/main/java/net/minecraft/server/ContainerCartography.java
+@@ -25,7 +25,6 @@ public class ContainerCartography extends Container {
+     }
+     // CraftBukkit end
+     private final ContainerAccess containerAccess;
+-    private boolean e;
```

Comment:
Keep the field there even if it is not used.

Reply:
It has been added back in